support-workers/src/test/scala/com/gu/zuora/subscriptionBuilders/DigitalSubscriptionBuilderSpec.scala

```diff
@@ -86,9 +102,19 @@ class DigitalSubscriptionBuilderSpec extends AsyncFlatSpec with Matchers {
   lazy val monthly = DigitalSubscriptionBuilder.build(
     DigitalPack(GBP, Monthly),
     UUID.fromString("f7651338-5d94-4f57-85fd-262030de9ad5"),
-    SubscriptionPaymentDirect(ZuoraDigitalPackConfig(14, 2), None, Country.UK, promotionService),
+    SubscriptionPurchase(ZuoraDigitalPackConfig(14, 2), None, Monthly, Country.UK, promotionService),
+    SANDBOX,
+    () => saleDate
+  ).value.map(_.right.get)
+
+  lazy val threeMonthGiftPurchase = DigitalSubscriptionBuilder.build(
+    DigitalPack(GBP, Quarterly, Gift),
+    UUID.fromString("f7651338-5d94-4f57-85fd-262030de9ad5"),
+    SubscriptionPurchase(ZuoraDigitalPackConfig(14, 2), None, Quarterly, Country.UK, promotionService),
     SANDBOX,
     () => saleDate
   ).value.map(_.right.get)
 
+  //TODO: RB test all cases
```

Comment:
GW gifting is a good one to check that we have the tests too, as I'd imagine that's not a super common code path.